sentry_sdk/integrations/fastapi.py

```diff
@@ -82,6 +83,60 @@ def _set_transaction_name_and_source(event, transaction_style, request):
     event["transaction_info"] = {"source": SOURCE_FOR_STYLE[transaction_style]}
 
 
+def patch_get_request_handler():
```

Comment:
if we patch the request handler, do we need `SentryFastApiMiddleware` anymore?

Reply:
same question for `SentryStarletteMiddleware`

Reply:
Right now the `SentryFastApiMiddleware` is only setting the transaction name and source. But I guess we can move this into the request handler and get rid of the middleware!

Reply:
yea I'd prefer only one place to set transaction name/source.

Reply:
There were some differences in Starlette and FastAPI. Can't remember exactly what, but something with routing or handling the routes. So I think we need to set the transaction name and source in both integrations to make sure we have the correct one in FastAPI projects. 
But we can get rid of the `SentryFastApiMiddleware`.

Reply:
yes I meant one place per Starlette/Fastapi integration.
So I think we can get rid of both `SentryFastApiMiddleware` and `SentryStarletteMiddleware` in theory.

Reply:
I have now cleaned up the code and the transaction name/source are only set once in fastapi and once in starlette integration.